src/driver/compile.ml

```diff
@@ -11,12 +11,14 @@ type pkg_args = {
 
 type compiled = {
   m : ty;
-  output_dir : Fpath.t;
-  output_file : Fpath.t;
+  odoc_output_dir : Fpath.t; (* e.g. "_odoc/base/lib/base/" *)
+  odoc_file : Fpath.t; (* Full path to odoc file *)
+  odocl_file : Fpath.t;
   include_dirs : Fpath.Set.t;
   impl : impl option;
   pkg_args : pkg_args;
-  package_name : string;
+  pkg_name : string;
+  pkg_dir : Fpath.t;
```

Comment:
In many places in this module, both of these values are computed from different sources. It would be nice to ensure that they are consistent by always computing them from a single source of truth:
```suggestion
  pkg : string * Fpath.t;
```

Reply:
I've made the changes.